content/BindingMethods/BindingMethods.q.md

```diff
@@ -0,0 +1,1656 @@
+
+- `1`
+
+- `2`
+
+- `$data[0]`
+
+- `$data[1]`
+
+- `PDO::PARAM_STR`
+
+:::
+
+
+:::
+
+/// type=REPL, filename=[connection.php,student.php]
+
+```php
+// connection.php
+<?php
+    $host = 'localhost';
+    $db = 'LibraryDB';
+    $port = '5432';
+    $username = 'codestop';
+    $password = 'Admin01';
+    try {
+        $conn = new PDO("pgsql:host=$host;port=$port;dbname=$db;user=$username;password=$password");
+        if ($conn) {
+            echo "Successfully connected to the database.\n";
+        }
+    } catch (Exception $e) {
+        echo "Unable to establish a connection."; 
+    }
+?>
+```
+
+```php
+// student.php
+<?php
+    require_once("connection.php");
+
+    $author_name = 'Naomi';
+    $sql = "SELECT * FROM authors WHERE first_name = :first_name";
+
+    try {
+        $pstmt = $conn->prepare($sql);
+        $pstmt->bindValue(':first_name', $author_name, PDO::PARAM_STR);
+        $pstmt->execute();
+        $data = $pstmt->fetchAll();
+        if (empty($data)) {
+            throw new Exception("Student record not found.");
+        }
+        foreach ($data as $row) {
+            echo $row['author_id']."\t".$row['first_name']."\t".$row['last_name']."\n";
+        }
+    } catch (Exception $e) {
+        echo $e->getMessage();
+    }
+?>
+```
+/// type=SS, answer=[4]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `1 Naomi Wolf` in separate lines.
+
+- It prints `Successfully connected to the database.` and `Student record not found.` in separate lines.
+
+
+/// type=SS, answer=[4]
+
+In the statement `$pstmt->bindValue(':first_name', $author_name, PDO::PARAM_STR);` on line 9 of `student.php`, replace the variable `$author_name` with the value `Naomi`. Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `1 Naomi Wolf` in separate lines.
+
+- It prints `Successfully connected to the database.` and `Student record not found.` in separate lines.
+
+:::
+
+
++++
+
+
++++
+
+### Part 2: Knowledge Assessment
+
+/// type=SS, answer=[1]
+
+Which statement best describes `PDOStatement::bindValue()`?
+
+- It binds a value to a parameter
+
+- It executes a prepared statement.
+
+- It binds a parameter to a variable.
+
+- It sets a proper data type in an SQL statement.
+
+- It sets a value to bind to the corresponding placeholder as a parameter.
+
+
+/// type=MS, answer=[2,3,4]
+
+Which statements correctly describe `PDOStatement::bindParam()`?
+
+- It executes a prepared statement.
+
+- It binds a parameter to a variable.
+
+- It sets a bounded variable as a reference.
+
+- It evaluates a bounded variable once the `execute()` method is called.
```

Comment:
Please check this statement here... Is it possible to match this with the statement in our facts? Thanks.